google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/AckRequestData.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub.v1;
+
+import com.google.api.core.SettableApiFuture;
+
+public class AckRequestData {
+  private final String ackId;
+  private final boolean exactlyOnceDeliveryEnabled;
+  private final boolean shouldSetMessageFutureOnSuccess;
+  private final SettableApiFuture<AckResponse> messageFuture;
+
+  protected AckRequestData(Builder builder) {
+    this.ackId = builder.ackId;
+    this.exactlyOnceDeliveryEnabled = builder.isExactlyOnceEnabled;
+    this.shouldSetMessageFutureOnSuccess = builder.shouldSetMessageFutureOnSuccess;
+    this.messageFuture =
```

Comment:
I would just make this an optional argument, because if we don't set a response on the future there is no point in creating a settableApiFuture if it will never complete

Reply:
Updated to use optional message future.